### PR TITLE
Fix skin result extraction based on list of elements

### LINKED ISSUE #423: Issues combining mesh skin with scoping
### Before submitting the issue

- [X] I have checked for [Compatibility](https://dpf.docs.pyansys.com/version/stable/getting_started/index.html#compatibility) issues
- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### Description of the bug

I'm playing with scoping in combination with skin mesh, and see something unexpected.
In this script
```python
from ansys.dpf import post

rst_path = r"D:\beam_2_mat.rst"

simulation = post.StaticMechanicalSimulation(rst_path)
simulation._model.metadata.meshed_region.plot()

# get an elemental named selection
scoping = simulation._model.metadata.meshed_region.named_selection(
    'X_MIN_SOLID_ELE'
)

seqv_ns_full = simulation.stress_eqv_von_mises_elemental(element_ids=scoping.ids)
seqv_ns_full.plot()

seqv_ns_skin = simulation.stress_eqv_von_mises_elemental(skin=scoping.ids)
seqv_ns_skin._fc[0].meshed_region.plot()

seqv_ns_skin.plot()
```

I get this (correct) plot for the field on the solid mesh
MicrosoftTeams-image

but this one on the skin
MicrosoftTeams-image (1)

The same rst file also contains other named selections that can be used for comparison. For instance `MG_BODY_ELE` and `MG_BODY_NODES` are named selections covering the same region with elemental and nodal location, respectively. The same applies to `ALU_BODY_ELE` and `ALU_BODY_NODES`.

Would be good to validate these kind of comparisons:

```python
# get an elemental named selection
scoping = simulation._model.metadata.meshed_region.named_selection(
    'MG_BODY_ELE'
)

seqv_ns_full = simulation.stress_eqv_von_mises_elemental(element_ids=scoping.ids)
seqv_ns_full.plot()

seqv_ns_skin = simulation.stress_eqv_von_mises_elemental(skin=scoping.ids)
seqv_ns_skin._fc[0].meshed_region.plot()
seqv_ns_skin.plot()

# get a nodal named selection
scoping = simulation._model.metadata.meshed_region.named_selection(
    'MG_BODY_NODES'
)

seqv_ns_full = simulation.stress_eqv_von_mises_nodal(node_ids=scoping.ids)
seqv_ns_full.plot()

seqv_ns_skin = simulation.stress_eqv_von_mises_nodal(skin=True, node_ids=scoping.ids)
seqv_ns_skin._fc[0].meshed_region.plot()
seqv_ns_skin.plot()
```

The rst file: [beam_2_mat.zip](https://github.com/ansys/pydpf-post/files/11818539/beam_2_mat.zip)

### Steps To Reproduce

-

### Which Operating System are you using?

Windows

### Which DPF/Ansys version are you using?

Ansys 2023 R2

### Which Python version are you using?

3.10

### Installed packages

-

### PR DESCRIPTION
**Pull request by PProfizi**
Closes #423

Fixes the initial issue while adding a new request argument for mesh reduction `reduce_mesh` to handle the behavior expected for remote post-processing:
- [ ] add tests
- [ ] confirm design and propagate to all result requests
- [ ] add example

### CODE REVIEW

File: tests/test_simulation.py [hunk 1]
```diff
@@ -2015,32 +2028,32 @@ def test_stress_skin(self, frame_modal_simulation: post.ModalMechanicalSimulatio
         )
         assert len(result.columns.set_ids) == 1
         if SERVERS_VERSION_GREATER_THAN_OR_EQUAL_TO_7_1:
-            assert len(result.index.mesh_index) == 36
```
Comment (cbellot000) [inline tests/test_simulation.py:2018]: how confident are we all these unit tests changes?

Comment (cbellot000) [reply tests/test_simulation.py:2018]: did you plot all of these to make sure they were good?

Comment (PProfizi) [reply tests/test_simulation.py:2018]: @cbellot000 Yes, I debugged most of them locally to check the actual number of elements involved and whether the new reference is correct this time. This is why I switched to NamedSelections for most of them.

Comment (cbellot000) [reply tests/test_simulation.py:2018]: I think we can now duplicate all these tests once with reduce_mesh=True, where we should have the same results as before, right?

File: src/ansys/dpf/post/static_mechanical_simulation.py [hunk 2]
```diff
@@ -1137,17 +1158,22 @@ def stress_eqv_von_mises_elemental(
...
-             of the mesh for plotting and data extraction. If a list is passed, the skin
-             is computed over list of elements (not supported for cyclic symmetry). Getting the
-             skin on more than one result (several time freq sets, split data...) is only
-             supported starting with Ansys 2023R2.
+            Select the skin (creates new 2D elements connecting the external nodes)
+            of the mesh for plotting and data extraction. If a list is passed, the skin
+            is computed over list of elements (not supported for cyclic symmetry). Getting the
+            skin on more than one result (several time freq sets, split data...) is only
+            supported starting with Ansys 2023R2.
+        reduce_mesh:
```
Comment (cbellot000) [inline src/ansys/dpf/post/static_mechanical_simulation.py:1172]: what about the external layer?
I'm not sure about the name "reduce" any other ideas, @rlagha @FedericoNegri?

File: src/ansys/dpf/post/static_mechanical_simulation.py [hunk 3]
```diff
@@ -1137,17 +1158,22 @@ def stress_eqv_von_mises_elemental(
...
+            layer is computed over list of elements.
         skin:
-             Select the skin (creates new 2D elements connecting the external nodes)
-             of the mesh for plotting and data extraction. If a list is passed, the skin
-             is computed over list of elements (not supported for cyclic symmetry). Getting the
-             skin on more than one result (several time freq sets, split data...) is only
-             supported starting with Ansys 2023R2.
+            Select the skin (creates new 2D elements connecting the external nodes)
+            of the mesh for plotting and data extraction. If a list is passed, the skin
+            is computed over list of elements (not supported for cyclic symmetry). Getting the
```
Comment (cbellot000) [inline src/ansys/dpf/post/static_mechanical_simulation.py:1169]: isn't this description true only for reduce_mesh==True?

File: tests/test_simulation.py (hunk 1 above)
Comment (PProfizi) [reply tests/test_simulation.py:2018]: @cbellot000 I started making specific tests for when we use `reduce_mesh`, but I do not think we need to duplicate all of the already existing ones for each result.

File: src/ansys/dpf/post/static_mechanical_simulation.py (hunk 2 above)
Comment (PProfizi) [reply src/ansys/dpf/post/static_mechanical_simulation.py:1172]: @cbellot000 you're right, I'll check what it does in combination with `external_layer`. As for the name `reduce_mesh`, it is definitely just a proposal.

File: src/ansys/dpf/post/static_mechanical_simulation.py (hunk 3 above)
Comment (PProfizi) [reply src/ansys/dpf/post/static_mechanical_simulation.py:1169]: @cbellot000 do you mean the fact that having `skin=[1, 2, 3]` will map results onto the skin of elements 1, 2, and 3? 

This is still true if `reduce_mesh=False`, where you will get results on "skin elements corresponding to elements 1, 2, and 3 based on the skin generated from the initial mesh". 

If `reduce_mesh=True`, as described in the docstring for it, you will indeed get results on the "all skin elements of the skin generated from the mesh composed of elements 1, 2, and 3 only".

File: src/ansys/dpf/post/selection.py [hunk 4]
```diff
@@ -798,6 +844,16 @@ def spatial_selection(self) -> SpatialSelection:
     def spatial_selection(self, value: SpatialSelection):
         self._spatial_selection = value
 
+    def reduce_mesh(self, element_ids: Union[List[int]]):
+        """Reduce the mesh under consideration to the given list of elements.
```
Comment (JennaPaikowsky) [inline src/ansys/dpf/post/selection.py:848]:
```suggestion
        """Reduce the mesh with consideration to the given list of elements.
```

File: src/ansys/dpf/post/static_mechanical_simulation.py [hunk 5]
```diff
@@ -104,6 +105,11 @@ def _get_result(
              is computed over list of elements (not supported for cyclic symmetry). Getting the
              skin on more than one result (several time freq sets, split data...) is only
              supported starting with Ansys 2023R2.
+        reduce_mesh:
+            Perform a reduction of the mesh under consideration based on, by order of priority:
+            - the list of element IDs given to this parameter
+            - parameter skin if reduce_mesh=True
+            - parameter element_ids if reduce_mesh=True and skin=None
```
Comment (JennaPaikowsky) [inline src/ansys/dpf/post/static_mechanical_simulation.py:112]:
```suggestion
            Perform a reduction of the mesh based on, by order of priority:
            - The list of element IDs given to this parameter
            - Parameter skin if reduce_mesh=True
            - Parameter element_ids if reduce_mesh=True and skin=None
```

File: src/ansys/dpf/post/static_mechanical_simulation.py [hunk 6]
```diff
@@ -1137,17 +1158,22 @@ def stress_eqv_von_mises_elemental(
...
-             Select the skin (creates new 2D elements connecting the external nodes)
-             of the mesh for plotting and data extraction. If a list is passed, the skin
-             is computed over list of elements (not supported for cyclic symmetry). Getting the
-             skin on more than one result (several time freq sets, split data...) is only
-             supported starting with Ansys 2023R2.
+            Select the skin (creates new 2D elements connecting the external nodes)
+            of the mesh for plotting and data extraction. If a list is passed, the skin
+            is computed over list of elements (not supported for cyclic symmetry). Getting the
+            skin on more than one result (several time freq sets, split data...) is only
+            supported starting with Ansys 2023R2.
```
Comment (JennaPaikowsky) [inline src/ansys/dpf/post/static_mechanical_simulation.py:1171]:
```suggestion
            supported starting with Ansys 2023 R2.
```

**Review by JennaPaikowsky — COMMENTED** [review]
Just a few minor changes

**Pull request closed without merging**